pkg/virt-controller/watch/vm.go

```diff
@@ -2615,9 +2616,32 @@ func (c *VMController) sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachin
 	// hotplugged volumes and interfaces
 	if c.needsSync(key) && syncErr == nil {
 		vmCopy := vm.DeepCopy()
-		if c.clusterConfig.HotplugNetworkInterfacesEnabled() {
-			if err = c.handleDynamicIfaceRequestOnVMI(vmCopy, vmi); err != nil {
-				syncErr = &syncErrorImpl{fmt.Errorf("Error encountered when trying to apply interface request on vmi: %v", err), HotPlugNetworkInterfaceErrorReason}
+		if c.clusterConfig.HotplugNetworkInterfacesEnabled() &&
+			vmi != nil && vmi.DeletionTimestamp == nil {
+			vmiCopy := vmi.DeepCopy()
+
+			indexedStatusIfaces := vmispec.IndexInterfacesFromStatus(vmi.Status.Interfaces,
+				func(ifaceStatus virtv1.VirtualMachineInstanceNetworkInterface) bool { return true })
+
+			ifaces, networks := clearDetachedInterfaces(vmCopy.Spec.Template.Spec.Domain.Devices.Interfaces, vmCopy.Spec.Template.Spec.Networks, indexedStatusIfaces)
+			vmCopy.Spec.Template.Spec.Domain.Devices.Interfaces = ifaces
+			vmCopy.Spec.Template.Spec.Networks = networks
+
+			ifaces, networks = clearDetachedInterfaces(vmiCopy.Spec.Domain.Devices.Interfaces, vmiCopy.Spec.Networks, indexedStatusIfaces)
+			vmiCopy.Spec.Domain.Devices.Interfaces = ifaces
+			vmiCopy.Spec.Networks = networks
+
+			if hasOrdinalIfaces, err := c.hasOrdinalNetworkInterfaces(vmi); err != nil {
```

Comment:
I think this can be done earlier to avoid processing.
it will also simplify the `syncErr` condition later.

But I can take this as a follow up.